src/interop-utils.ts

```diff
@@ -0,0 +1,142 @@
+// src/interop-utils.ts
+import {ethers, JsonRpcProvider} from 'ethers';
+import {utils, types} from './index';
+import {Provider} from './provider';
+
+/**
+ * Locate the L1Messenger service log inside a receipt.
+ *
+ * Matches by:
+ * - `sender == L1_MESSENGER_ADDRESS`
+ * - `key == zeroPad(eoa, 32)`
+ * - optional `value == keccak256(message)` if `message` is provided
+ *
+ * @internal
+ * @param receipt - Source L2 tx receipt with `l2ToL1Logs`.
+ * @param eoa - Externally owned account (sender) to match.
+ * @param message - Optional raw message bytes to match by value hash.
+ * @returns The log index within `receipt.l2ToL1Logs`, or `-1` if not found.
+ */
+export function findInteropLogIndex(
+  receipt: types.TransactionReceipt,
+  eoa: types.Address,
```

Comment:
maybe rename eoa to sender as if a contract sends the message then this value is the contract address?

Reply:
Good call!